src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -33,86 +28,161 @@
     private EnvVars env;
-    private boolean tapChkBx;
-    private boolean junitChkBx;
-    private boolean coberturaChkBx;
-    private boolean stmResultsChkBx;
-    private boolean modelCoverageChkBx;
-    private boolean pdfReportChkBx;
+    private TapChkBx tapChkBx;
+    private JunitChkBx junitChkBx;   
+    private CoberturaChkBx coberturaChkBx;
+    private StmResultsChkBx stmResultsChkBx; 
+    private ModelCovChkBx modelCoverageChkBx; 
+    private PdfChkBx pdfReportChkBx;
+    private String tapReportFilePath;
+    private String pdfReportFilePath;
+    private String junitReportFilePath;
+    private String coberturaReportFilePath;
+    private String stmResultsFilePath;
+    private String modelCoverageFilePath;
+   
+    private List<String> inputArgs = new ArrayList<String>();
 
     @DataBoundConstructor
     public RunMatlabTestsBuilder() {
 
-
     }
 
 
     // Getter and Setters to access local members
 
 
     @DataBoundSetter
-    public void setTapChkBx(boolean tapChkBx) {
+    public void setTapChkBx(TapChkBx tapChkBx) {
         this.tapChkBx = tapChkBx;
-    }
-
+        this.tapReportFilePath = this.tapChkBx.getTapReportFilePath();
+    } 
+   
     @DataBoundSetter
-    public void setJunitChkBx(boolean junitChkBx) {
+    public void setJunitChkBx(JunitChkBx junitChkBx) {
         this.junitChkBx = junitChkBx;
+        this.junitReportFilePath = this.junitChkBx.getJunitReportFilePath();
     }
-
+    
     @DataBoundSetter
-    public void setCoberturaChkBx(boolean coberturaChkBx) {
+    public void setCoberturaChkBx(CoberturaChkBx coberturaChkBx) {
         this.coberturaChkBx = coberturaChkBx;
+        this.coberturaReportFilePath = this.coberturaChkBx.getCoberturaReportFilePath();
     }
-
+    
     @DataBoundSetter
-    public void setStmResultsChkBx(boolean stmResultsChkBx) {
+    public void setStmResultsChkBx(StmResultsChkBx stmResultsChkBx) {
         this.stmResultsChkBx = stmResultsChkBx;
+        this.stmResultsFilePath = this.stmResultsChkBx.getStmResultsFilePath();
     }
-
+    
     @DataBoundSetter
-    public void setModelCoverageChkBx(boolean modelCoverageChkBx) {
+    public void setModelCoverageChkBx(ModelCovChkBx modelCoverageChkBx) {
         this.modelCoverageChkBx = modelCoverageChkBx;
-    }
+        this.modelCoverageFilePath = this.modelCoverageChkBx.getModelCoverageFilePath();
+    }   
 
     @DataBoundSetter
-    public void setPdfReportChkBx(boolean pdfReportChkBx) {
+    public void setPdfReportChkBx(PdfChkBx pdfReportChkBx) {
         this.pdfReportChkBx = pdfReportChkBx;
+        this.pdfReportFilePath = this.pdfReportChkBx.getPdfReportFilePath();
     }
-
-    public boolean getTapChkBx() {
-        return tapChkBx;
+    
+    public String getTapReportFilePath() {
+        return this.tapReportFilePath;
+    }      
+    public boolean getIsPdfChecked() {
+        if(this.pdfReportChkBx != null) {
+            return true;
+        }
+        return false;

```

Comment:
Does null always mean false? I am thinking of the case where the check box was checked, the. alter unchecked. Is this stateful such that it is unchecked and thus should be false? Worried about bugs here with this logic. If there is no bug, this can be simplified to:

    return this.pdfReportChkBx != null

...but really I prefer the approach of using the NullObject pattern and avoiding the raw nulls entirely. See a later comment.

Reply:
Yes, NULL here always means false. I have already removed all these methods i could achive the same in jelly by inline expressions instead.